From 07effbfa34c46132786a28f1ff3fa3f1ce49fbd6 Mon Sep 17 00:00:00 2001
From: Lei Xu <lei@apache.org>
Date: Fri, 2 Sep 2016 11:01:13 -0700
Subject: [PATCH 1835/2848] HDFS-10822. Log DataNodes in the write pipeline.
 John Zhuge via Lei Xu

(cherry picked from commit 5a8c5064d1a1d596b1f5c385299a86ec6ab9ad1e)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DataStreamer.java

Change-Id: I283eec9af975b83259ad1330d3a15346599b299a
---
 .../org/apache/hadoop/hdfs/DFSOutputStream.java    |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java
index bb54039..a10c9fd 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java
@@ -588,6 +588,11 @@ private void setFavoredNodes(String[] favoredNodes) {
     private void initDataStreaming() {
       this.setName("DataStreamer for file " + src +
           " block " + block);
+      if (DFSClient.LOG.isDebugEnabled()) {
+        DFSClient.LOG.debug("nodes " + Arrays.toString(nodes) +
+                " storageTypes " + Arrays.toString(storageTypes) +
+                " storageIDs " + Arrays.toString(storageIDs));
+      }
       response = new ResponseProcessor(nodes);
       response.start();
       stage = BlockConstructionStage.DATA_STREAMING;
-- 
1.7.9.5

